src/server/scripts/EasternKingdoms/Stratholme/instance_stratholme.cpp

```diff
@@ -214,14 +227,41 @@ class instance_stratholme : public InstanceMapScript
                                 break;
                             case FAIL:
                                 DoRemoveAurasDueToSpellOnPlayers(SPELL_BARON_ULTIMATUM);
+                                if (Creature* ysida = instance->GetCreature(ysidaGUID))
+                                    ysida->CastSpell(ysida, SPELL_PERM_FEIGN_DEATH, true);
                                 EncounterState[0] = data;
                                 break;
                             case DONE:
                                 EncounterState[0] = data;
-                                if (Creature* ysidaTrigger = instance->GetCreature(ysidaTriggerGUID))
+
+                                if (Creature* ysida = instance->GetCreature(ysidaGUID))
                                 {
-                                    Position ysidaPos = ysidaTrigger->GetPosition();
-                                    ysidaTrigger->SummonCreature(NPC_YSIDA, ysidaPos, TEMPSUMMON_TIMED_DESPAWN, 1800000);
+                                    if (GameObject* cage = instance->GetGameObject(ysidaCageGUID))
+                                        cage->UseDoorOrButton();
+
+                                    float x, y, z;
+                                    //! This spell handles the Dead man's plea quest completion
+                                    ysida->CastSpell(nullptr, SPELL_YSIDA_SAVED, true);
```

Comment:
i remember that treeston made some stuff to avoid this CastSpell(nullptr...

Reply:
https://github.com/TrinityCore/TrinityCore/commit/c7896f31026d6bdee189226680705dd7965b2039